src/containers/blocks.jsx

```diff
@@ -264,4 +298,17 @@ Blocks.defaultProps = {
     options: Blocks.defaultOptions
 };
 
-export default Blocks;
+const mapStateToProps = state => ({
+    toolboxXML: state.toolbox.toolboxXML
+});
+
+const mapDispatchToProps = dispatch => ({
+    sendToolboxUpdate: toolboxXML => {
```

Comment:
I think the convention for dispatch props is to name them `onSomethingSomething`. The naming helps to keep track of where things come from. In the same vein, I think `onExtensionWasAdded` should be renamed to `handleExtensionAdded` to be consistent with our other naming (`handleSomething` vs. `onSomething`)